eo-maven-plugin/src/main/java/org/eolang/maven/RegisterMojo.java

```diff
@@ -61,12 +64,19 @@ public final class RegisterMojo extends SafeMojo {
     private boolean strictFileNames = true;
 
     @Override
-    public void exec() {
+    public void exec() throws IOException {
         if (this.sourcesDir == null) {
             throw new IllegalArgumentException(
                 String.format("sourcesDir is null. Please specify a valid sourcesDir for %s", this)
             );
         }
+        if (this.foreign.exists()) {
+            Files.delete(this.foreign.toPath());
+        }
+        final File pulled = this.targetDir.toPath().resolve(PullMojo.DIR).toFile();
+        if (pulled.exists()) {
+            new CleanFiles(pulled).clean();
+        }
```

Comment:
@Suban05 I think we should clear not only pulled files but also resolved ones, check the [ticket](https://github.com/objectionary/eo/issues/3503#issuecomment-2481390264) carefully